lib/charms/haproxy/v1/haproxy_route.py

```diff
@@ -303,18 +303,42 @@ class ServerHealthCheck(BaseModel):
         port: Customize port value for http-check.
     """
 
-    interval: int = Field(
-        description="The interval (in seconds) between health checks.", default=60
+    interval: Optional[int] = Field(
+        description="The interval (in seconds) between health checks.", default=None
     )
-    rise: int = Field(
-        description="How many successful health checks before server is considered up.", default=2
+    rise: Optional[int] = Field(
+        description="How many successful health checks before server is considered up.",
+        default=None,
     )
-    fall: int = Field(
-        description="How many failed health checks before server is considered down.", default=3
+    fall: Optional[int] = Field(
+        description="How many failed health checks before server is considered down.", default=None
     )
     path: Optional[VALIDSTR] = Field(description="The health check path.", default=None)
     port: Optional[int] = Field(description="The health check port.", default=None)
 
+    @model_validator(mode="after")
+    def check_all_required_fields_set(self) -> Self:
+        """Check that all required fields for health check are set.
+
+        Raises:
+            ValueError: When validation fails.
+
+        Returns:
+            The validated model.
+        """
+        if not (bool(self.interval) == bool(self.rise) == bool(self.fall)):
+            raise ValueError("All three of interval, rise and fall must be set.")
+        return self
+
+    @property
+    def is_health_check_configured(self) -> bool:
+        """Indicate if the backend has activated health checks.
+
+        Returns:
+            bool: Whether health check has been configured.
+        """
+        return sum(bool(value is not None) for value in [self.interval, self.rise, self.fall]) == 3
```

Comment:
I think using `all` is cleaner
```suggestion
        return all([self.interval, self.rise, self.fall])
```

Reply:
This was redundant, I removed it.